<launch>

  <arg name="publisher_uav_node_id" default="97"/>
  <arg name="subscriber_uav_node_id" default="98"/>
  <arg name="uav_can_interface" default="can0"/>

  <node name="ros_to_uavcan_bridge_node" pkg="uavcan_ros_bridge" type="ros_to_uavcan_bridge" respawn="true" output="screen">
    <param name="uav_node_id" value="$(arg subscriber_uav_node_id)"/>
    <param name="uav_can_interface" value="$(arg uav_can_interface)"/>
    <remap from="~command" to="uavcan_command"/>
    <remap from="~array_command" to="uavcan_array_command"/>
    <remap from="~vbs_command" to="uavcan_vbs_command"/>
    <remap from="~tcg_command" to="uavcan_tcg_command"/>
    <remap from="~lcg_command" to="uavcan_lcg_command"/>
  </node>

  <node name="uavcan_to_ros_bridge_node" pkg="uavcan_ros_bridge" type="uavcan_to_ros_bridge" respawn="false" output="screen">
    <param name="uav_node_id" value="$(arg publisher_uav_node_id)"/>
    <param name="uav_can_interface" value="$(arg uav_can_interface)"/>
    <remap from="~imu" to="uavcan_imu"/>
    <remap from="~pressure1" to="uavcan_pressure1"/>
    <remap from="~pressure2" to="uavcan_pressure2"/>
  </node>

  <!--
  <node name="ros_to_uavcan_services_node" pkg="uavcan_ros_bridge" type="ros_to_uavcan_services" respawn="true" output="screen">
    <param name="uav_node_id" value="99"/>
    <param name="uav_can_interface" value="$(arg uav_can_interface)"/>
    <remap from="~get_node_info" to="uavcan_get_node_info"/>
    <remap from="~get_transport_stats" to="uavcan_get_transport_stats"/>
  </node>
  -->

</launch>
